Source/WebKitLegacy/WebCoreSupport/WebCryptoClient.mm

```diff
@@ -50,21 +51,25 @@
     return wrappedKey;
 }
 
-std::optional<Vector<uint8_t>> WebCryptoClient::unwrapCryptoKey(const Vector<uint8_t>& wrappedKey) const
+std::optional<Vector<uint8_t>> WebCryptoClient::unwrapCryptoKey(const Vector<uint8_t>& serializedKey) const
 {
+    auto wrappedKey = WebCore::readSerializedCryptoKey(serializedKey);
+    if (!wrappedKey)
+        return std::nullopt;
     SEL selector = @selector(webCryptoMasterKeyForWebView:);
-    Vector<uint8_t> key;
     if ([[m_webView UIDelegate] respondsToSelector:selector]) {
         auto masterKey = makeVector(CallUIDelegate(m_webView, selector));
-        if (!WebCore::unwrapSerializedCryptoKey(masterKey, wrappedKey, key))
+        auto key = WebCore::unwrapCryptoKey(masterKey, *wrappedKey);
+        if (!key)
             return std::nullopt;
         return key;
```

Comment:
It's a bit more idiomatic (closer to WebKit style) to reverse these two lines (and since getting a key is the more likely code path):

```suggestion
        if (auto key = WebCore::unwrapCryptoKey(masterKey, *wrappedKey))
            return key;
        return std::nullopt;
```

I think this also works, but please check first:

```suggestion
        return WebCore::unwrapCryptoKey(masterKey, *wrappedKey) ?: std::nullopt;
```

Or if `WebCore::unwrapCryptoKey()` already returns `std::nullopt` on error, just:

```suggestion
        return WebCore::unwrapCryptoKey(masterKey, *wrappedKey);
```

Reply:
Excellent suggestion!!!